contracts/collections/src/contract.rs

```diff
@@ -213,7 +213,10 @@ impl Collections {
         id: u64,
         transfer_amount: u64,
     ) -> Result<(), ContractError> {
-        Self::is_authorized_for_transfer(&env, &sender)?;
+        let can_do_all = Self::is_authorized_for_all(&env, &sender);
+        if can_do_all.is_err() {
+            Self::is_authorized_for_transfer(&env, &sender)?;
+        }
```

Comment:
This is redundant since `is_authorized_for_transfer` already checks `_for_all`.

Reply:
removed